examples/jsm/exporters/EXRExporter.js

```diff
@@ -205,7 +205,7 @@ async function getPixelData( renderer, rtt, info ) {
 
 		}
 
-		renderer.readRenderTargetPixels( rtt, 0, 0, info.width, info.height, dataBuffer );
+		await renderer.readRenderTargetPixelsAsync( rtt, 0, 0, info.width, info.height, dataBuffer );
```

Comment:
I notice that the function signature for `readRenderTargetPixelsAsync` in WebGPURenderer doesn't take a `dataBuffer` argument which should probably be fixed to align with the WebGLRenderer implementation. Taking a databuffer was a deliberate decision to avoid memory allocation in cases where this is run ever frame or every few frames and is common pattern use throughout the library, as well, and I think it's worth keeping.

It looks like WebGPURenderer will now create a new typed array on every call.

Reply:
To you mind updating `KTX2Exporter` as well?

Reply:
I've updated KTX2Exporter but haven't tested it since I don't see an example.

Also is it worth creating a new issue to track this function signature difference?

Reply:
The WebGPU allocates the buffer returned internally, we just pass the required size etc. We effectively cast it to the correct type on return, but can't provide a buffer to be used, I think to ensure that the buffer alignment and location with memory is suitable for efficient CPU <-> GPU transfers.

The buffer is created at https://github.com/mrdoob/three.js/blob/3b84981a3dfa81fc5462d0e61ad9a1e3848225be/src/renderers/webgpu/utils/WebGPUTextureUtils.js#L384

And made available to JS here:
 https://github.com/mrdoob/three.js/blob/3b84981a3dfa81fc5462d0e61ad9a1e3848225be/src/renderers/webgpu/utils/WebGPUTextureUtils.js#L415

Reply:
> Also is it worth creating a new issue to track this function signature difference?

Sounds good. Then we have a dedicated place to discuss this topic.

Reply:
If there is a use case that requires constant readback the recommended method would be to create a pool of WebGPU buffer objects and in a ring buffer arrangement, something that doesn't match the current renderer API.

Reply:
I have also realized that `WebGPURenderer.readRenderTargetPixelsAsync()` can not handle cube maps so far since you can't assign an active cube face index. This is required for a bunch of use cases like e.g. in `LightProbeGenerator`.

Reply:
I'll take a look.

Reply:
Relevant line:

https://github.com/mrdoob/three.js/blob/ce180c97303bb0606f3e314bd0d803ae5eb358fe/examples/jsm/lights/LightProbeGenerator.js#L165